adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenRequestTest.java

```diff
@@ -554,6 +559,306 @@ public void testVerifyBrokerRedirectUriInvalidSignature()
+                VALID_AUTHORITY, false, cacheStore);
+        //set useBroker into false
+        AuthenticationSettings.INSTANCE.setUseBroker(false);
+        //turn on the outage mode
+        authContext.setExtendedLifetimeEnabled(false);
+        //mock HttpUrlConnection for refresh token request
+        prepareSuccessHttpUrlConnection();
+
+        try {
+            final AuthenticationResult result = authContext.acquireTokenSilentSync("resource", "clientid", TEST_USERID);
+            assertNotNull(result);
+            assertTrue(result.getAccessToken().equals("I am an AT"));
+        } catch (final AuthenticationException exception) {
+            fail("Not expected exception.");
+        } finally {
+            cacheStore.removeAll();
+        }
+    }
+
+    /**
+     * Test for throwing exception when the request is rejected by server through there is a valid 
+     * stale AT in the cache and the outage mode is on.
+     */
+    @SmallTest
+    public void testResiliencyTokenReturn_outageModeOn_noRetry() throws PackageManager.NameNotFoundException,
+            NoSuchAlgorithmException, OperationCanceledException, IOException, AuthenticatorException,
+            InterruptedException {
+        // make sure AT's expires_in is expired and ext_expires_in is not expired
+        final Calendar expiredTime = new GregorianCalendar();
+        expiredTime.add(Calendar.MINUTE, -MINUS_MINUITE);
+        final Calendar extendedExpiresTime = new GregorianCalendar();
+        extendedExpiresTime.add(Calendar.MINUTE, EXTEND_MINUS_MINUTE);
+        final ITokenCacheStore cacheStore = getTokenCache(expiredTime.getTime(), false, false, extendedExpiresTime.getTime());
+
+        final FileMockContext mockContext = createMockContext();
+        AuthenticationSettings.INSTANCE.setUseBroker(false);
+        final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                VALID_AUTHORITY, false, cacheStore);
+        //turn on the outage mode
+        authContext.setExtendedLifetimeEnabled(true);
+        final TestAuthCallback callback = new TestAuthCallback();
+
+        //mock http response
+        final HttpURLConnection mockedConnection = Mockito.mock(HttpURLConnection.class);
+        HttpUrlConnectionFactory.mockedConnection = mockedConnection;
+        Util.prepareMockedUrlConnection(mockedConnection);
+        Mockito.when(mockedConnection.getOutputStream()).thenReturn(Mockito.mock(OutputStream.class));
+        Mockito.when(mockedConnection.getInputStream()).thenReturn(Util.createInputStream(Util.getSuccessTokenResponse(false, false)));
+        Mockito.when(mockedConnection.getResponseCode()).thenReturn(HttpURLConnection.HTTP_BAD_GATEWAY);
+
+        try {
+            authContext.acquireTokenSilentSync("resource", "clientid", TEST_USERID);
+            fail("Not expected exception.");
+        } catch (final Exception exception) {
```

Comment:
Because I want to verify the exception is an instance of AuthenticationException here. However, if I do catch(AuthenticationException exception), it will be meaningless to do the verification below, I think.